docs/proposals/node-agent/agent.proto

```diff
@@ -0,0 +1,41 @@
+syntax = "proto3";
+
+package node_agent;
+
+enum WorkloadStatus {
+    WAITING = 0;
+    RUNNING = 1;
+    TERMINATED = 2;
+}
+
+message WorkloadRequest {
+    string id = 1;
+}
+
+enum WorkloadType {
+    CONTAINER = 0;
+}
+
+message WorkloadResource {
+    int32 cpu = 1;
+    int32 memory = 2;
+    int32 disk = 3;
+}
+
+message WorkloadDefinition {
+    string id = 1;
+    string name = 2;
+    WorkloadType type = 3;
+    string image = 4;
+    WorkloadStatus status = 5;
+    repeated string environment = 6;
+    WorkloadResource resources = 7;
+}
+
+message Empty {}
+
+service WorkloadService {
+    rpc Create (WorkloadDefinition) returns (Empty) {}
+    rpc Delete (WorkloadDefinition) returns (Empty) {}
+    rpc Status (WorkloadDefinition) returns (WorkloadDefinition) {}
```

Comment:
Please view comment on the `README` about sending status updates.

Reply:
Fixed here : 717e49651e9b38a4eb925fe3015790fff71261f4

There is a Workload status type meant to be sent using the interface you proposed. Once we're set on my side, we should talk about how to merge our proto files.